hostports/watcher.go

```diff
@@ -109,6 +112,25 @@ func (p PortRule) natIptables() []byte {
 	return buf.Bytes()
 }
 
+func (p ChainRule) filterIptables() []byte {
+	buf := &bytes.Buffer{}
+	dockerBridge := os.Getenv("DOCKER_BRIDGE")
+	if p.Bridge != dockerBridge {
+		// bridged
+		metadataIP := os.Getenv("METADATA_IP")
+		rule := fmt.Sprintf("-A CATTLE_FORWARD -i %s -o %s -d %s -j ACCEPT\n", p.Bridge, p.Bridge, p.BridgeSubnet)
+		buf.WriteString(rule)
+		rule = fmt.Sprintf("-A CATTLE_FORWARD -i %s -o %s -d %s -j ACCEPT\n", p.Bridge, dockerBridge, metadataIP)
```

Comment:
You need similar rules in nat table as well. Otherwise the container's source IP address is masqueraded while accessing metadata. This is if the dockerBridge and cniBridge are different.

This should be the first rule in the chain `CATTLE_NAT_POSTROUTING`
```
-A CATTLE_NAT_POSTROUTING -d 169.254.169.250 -s 10.42.0.0/16 -j ACCEPT
```